10-ConcurrencyAndSynchronization/mpu6050.c

```diff
@@ -195,10 +272,39 @@ CLASS_ATTR_RO(temperature);
 
 static struct class *attr_class;
 
+int bg_thread(void *data)
+{
+	pr_info("%s started", __func__);
+	while (!kthread_should_stop()) {
+		/**check requirement for*/
+		// pr_info("bg_thread check flag");
+		if (atomic_read(&update_req) != 0) {
+			pr_info("%s executing", __func__);
+			mpu6050_exec_reading();
+			atomic_set(&update_req, 0);
+			pr_info("reset update_req and read_done");
+			complete(&read_done);
+		}
+		schedule();
+	}
+	pr_info("sub_task stoped\n");
+	sub_task = NULL;
+	return 0;
+}
+
 static int mpu6050_init(void)
 {
 	int ret;
 
+	sub_task = kthread_create(bg_thread, NULL, "mpu6050_sub_task");
+	if (!IS_ERR(sub_task)) {
+		if (wake_up_process(sub_task))
```

Comment:
For the first time `wake_up_new_task()` should be used.

Reply:
I can say that this site was seen in
https://lxr.missinglinkelectronics.com/linux+v4.13/include/linux/kthread.h#L33